src/analysis/retail/shaman/shared/hero/stormbringer/TempestGraph.tsx

```diff
@@ -0,0 +1,175 @@
+  const tooltip: StringFieldDef<any> = {
+    field: 'value',
+    type: 'quantitative',
+  };
+
+  const spec: VisualizationSpec = {
+    layer: [
+      /** Maelstrom progress */
+      {
+        data: {
+          name: 'maelstrom',
+        },
+        mark: {
+          type: 'area',
+          point: point,
+          line: {
+            interpolate: interpolate,
+            color: colors.maelstrom,
+            strokeWidth: strokeWidth,
+            opacity: 1,
+          },
+          interpolate: interpolate,
+          opacity: 0.1,
+        },
+        transform: transforms,
+        encoding: {
+          ...baseEncoding,
+          y: {
+            field: 'value',
+            type: 'quantitative',
+            title: 'Maelstrom',
+            axis: {
+              grid: false,
+              format: '~s',
+              orient: 'left',
+              tickCount: info.combatant.spec === SPECS.ENHANCEMENT_SHAMAN ? 4 : 6,
+            },
+          },
+          tooltip: [
+            {
+              ...tooltip,
+              title: 'Maelstrom',
+            },
+          ],
+          color: {
+            value: colors.maelstrom,
+          },
+        },
+      },
+
+      /** Awakening Storms stacks */
+      {
+        data: {
+          name: 'awakeningStorms',
+        },
+        mark: {
+          type: 'area',
+          point: point,
+          line: {
+            color: colors.awakeningStorms,
+            interpolate: interpolate,
+            opacity: 1,
+            strokeWidth: strokeWidth,
+          },
+          opacity: 0.1,
+          interpolate: interpolate,
+        },
+        transform: transforms,
+        encoding: {
+          ...baseEncoding,
+          y: {
+            field: 'value',
+            type: 'quantitative',
+            title: 'Awakening Storms',
+            axis: {
+              orient: 'right',
+              tickCount: 3,
+            },
+          },
+          tooltip: [
+            {
+              ...tooltip,
+              title: 'Awakening Storms stacks',
+            },
+          ],
+          color: { value: colors.awakeningStorms },
+        },
+      },
+      /** Tempest casts */
+      {
+        data: {
+          name: 'tempest',
+        },
+        mark: {
+          type: 'rule',
+          color: '#fab700',
+          strokeWidth: 2,
+          opacity: 1,
+        },
+        encoding: {
+          x: baseEncoding.x,
+          tooltip: [
+            {
+              field: 'timestamp_shifted',
+              type: 'quantitative',
+              title: 'Tempest Cast',
+              formatType: 'timestamp',
+              format: '3',
+            },
+          ],
+        },
+        transform: transforms,
+      },
+    ],
+    resolve: {
+      scale: {
+        y: 'independent',
+      },
+    },
+  };
+
+  return spec;
+};
+
+expressionFunction('timestamp', function (datum: number, params: string) {
```

Comment:
this is neat. im gonna steal this

Reply:
I realised just now that I can replace the body with a call to `formatDuration`